src/associations/associations.service.ts

```diff
@@ -19,56 +18,35 @@ export class AssociationsService implements OnModuleInit {
     this.transactionTypes = [16, 17];
   }
 
-  onModuleInit(): void {
-    this.root = this.config.getAssociationsRoot();
-  }
-
-  async index(index: IndexDocumentType) {
+  async index(index: IndexDocumentType): Promise<void> {
     const { transaction } = index;
+    const { sender } = transaction;
 
     if (this.transactionTypes.indexOf(transaction.type) === -1){
-      return ;
-    }
-
-    this.logger.debug(`association-service: Indexing association`);
-
-    if (transaction.type === 16) {
-      return this.createAssocIndex(transaction);
-    } else if (transaction.type === 17) {
-      return this.removeAssocIndex(transaction);
+      this.logger.debug(`association-service: Unknown transaction type`);
+      return;
     }
-  }
-
-  async createAssocIndex(transaction: Transaction) {
 
-    const {sender, associationType} = transaction;
+    const associationIndexing = this.config.getAssociationIndexing();
+    const senderRoles = await this.storage.getRolesFor(sender);
+    const isSenderTrustNetwork = Object.keys(senderRoles).length > 0;
```

Comment:
Move this into the `if`, so this is only executed if indexing is set to 'trust'.

Reply:
done